KsApi/Service.swift

```diff
@@ -790,4 +790,13 @@ public struct Service: ServiceType {
       .perform(mutation: GraphAPI.WatchProjectMutation(input: GraphAPI.WatchProjectInput.from(input)))
       .flatMap(WatchProjectResponseEnvelope.producer(from:))
   }
+
+  public func blockUser(userId: String)
+    -> SignalProducer<EmptyResponseEnvelope, ErrorEnvelope> {
+    return GraphQL.shared.client
+      .perform(mutation: GraphAPI.BlockUserMutation(input: GraphAPI.BlockUserInput(blockUserId: userId)))
+      .flatMap { _ in
+        SignalProducer(value: EmptyResponseEnvelope())
```

Comment:
So `BlockUser` _can_ return two fields: `success` and `currentUser { blockedUsers } `. However, based on my current understanding of the client designs, we don't need to read either of these fields: we'll fetch `isBlocked` directly on other `User` objects.

Is that correct? Based on that assumption I figured I'd take the simplest route out and return an `EmptyResponseEnvelope`.

Reply:
Yep exactly. We only need to know if it was successful right now.